includes/multi-currency/class-multi-currency.php

```diff
@@ -187,14 +237,15 @@ public function get_enabled_currencies() {
 			return $this->enabled_currencies;
 		}
 
-		$this->enabled_currencies = get_option( $this->id . '_enabled_currencies', false );
-		if ( ! $this->enabled_currencies ) {
+		$available_currencies = $this->get_available_currencies();
```

Comment:
Usually I would expect a getter function to return the current value of a property without modifying the value of the property or any other properties in the class (so that the function has a single responsibility). What do you think about making the `get_*` functions simply return the value of the property, and then extracting the logic you have which sets up the initial values for `default_currency`, `available_currencies` and `enabled_currencies` to new methods?

In the case of `default_currency`, I think we could just have a simple `set_default_currency( $currency )` function which we call in the `init()` method (line 96) like so:

```php
$this->set_default_currency( $this->available_currencies[ get_woocommerce_currency() ] );
```

In the case of `available_currencies` and `enabled_currencies` I think it would be good to have a function called something like `initialise_available_currencies()` and `initialise_enabled_currencies()` which we call in the `init()` function where we currently call `get_enabled_currencies()` and `get_available_currencies()`.

These functions could contain all of the initial setup logic for the currencies which you currently have in the `get_enabled_currencies()` and `get_available_currencies()` functions.

Let me know what you think! 🙂

Reply:
Addressed this issue here: https://github.com/Automattic/woocommerce-payments/pull/2006
I chose as a PR rather than a direct commit, as I wanted input on the changes before bringing it in.